tests/test_calculator.cc

```diff
@@ -311,6 +314,102 @@ namespace rascal {
     }
   }
 
+  /* ---------------------------------------------------------------------- */
+
+  using grad_sparse_fixtures =
+      boost::mpl::list<CalculatorFixture<ComplexHypersSphericalInvariants>>;
+
+  /**
+   * Test if the keys for the PowerSpectrum's gradient are properly sparse on
+   * the 1st center of CaCrP2O7_mvc-11955_symmetrized.json
+   */
+  BOOST_FIXTURE_TEST_CASE_TEMPLATE(sparse_grad_test, Fix, grad_sparse_fixtures,
+                                   Fix) {
+    auto & managers = Fix::managers;
+    auto & representations = Fix::representations;
+    using Representation_t = typename Fix::Representation_t;
+    using PropGrad_t = typename Representation_t::template PropertyGradient_t<
+        typename Fix::Manager_t>;
+    auto & hypers = Fix::representation_hypers;
+    const bool verbose{false};
+    // assumes the structure is CaCrP2O7_mvc-11955_symmetrized.json
+    // the keys of the center contribution (1st center)
+    std::vector<std::vector<int>> all_keys{{8, 8},   {8, 15},  {8, 20},
+                                           {15, 15}, {15, 20}, {20, 20}};
+    // the list of keys for each neighbor (1st center)
+    std::vector<std::vector<std::vector<int>>> neigh_keys{
+        {{8, 15}, {15, 15}, {15, 20}}, {{8, 8}, {8, 15}, {8, 20}},
+        {{8, 8}, {8, 15}, {8, 20}},    {{8, 8}, {8, 15}, {8, 20}},
```

Comment:
Formatting is weird here. Would be clearer like this:

```
        {{8, 15}, {15, 15}, {15, 20}}, 
        {{8, 8}, {8, 15}, {8, 20}},
        {{8, 8},  {8, 15},  {8, 20}},  
        {{8, 8}, {8, 15}, {8, 20}},
```

Or without the spaces inside. Or just let clang-format do its thing =)

Reply:
this is the auto format